File: backend/routers/v1/cicd.go
package v1

import (
	cicdAPI "devops-platform/internal/modules/cicd/api"
	"devops-platform/internal/middleware"

	"github.com/gin-gonic/gin"
)

func registerCICD(r *gin.RouterGroup) {
	g := r.Group("/cicd")
	queryPermission := middleware.RequirePermission("cicd", "list")
	updatePermission := middleware.RequirePermission("cicd", "update")

	// Jenkins config
	g.GET("/jenkins", queryPermission, cicdAPI.ListJenkinsConfigs)
	g.POST("/jenkins", updatePermission,
		middleware.SetAuditOperation("Jenkins 配置保存"),
		cicdAPI.SaveJenkinsConfig)
	g.PUT("/jenkins/:id", updatePermission,
		middleware.SetAuditOperation("Jenkins 配置更新"),
		cicdAPI.SaveJenkinsConfig)
	g.DELETE("/jenkins/:id", updatePermission,
		middleware.SetAuditOperation("Jenkins 配置删除"),
		cicdAPI.DeleteJenkinsConfig)
	g.POST("/jenkins/test", queryPermission, cicdAPI.TestJenkinsConnection)

	// Jobs & builds
	g.GET("/jenkins/:configId/jobs", queryPermission, cicdAPI.ListJobs)
	g.POST("/jenkins/:configId/build", updatePermission,
		middleware.SetAuditOperation("触发 Jenkins 构建"),
		cicdAPI.TriggerBuild)
	g.GET("/jenkins/:configId/builds", queryPermission, cicdAPI.ListBuilds)
	g.GET("/jenkins/:configId/build-log", queryPermission, cicdAPI.GetBuildLog)

	// Pipelines
	g.GET("/pipelines", queryPermission, cicdAPI.ListPipelines)
	g.POST("/pipelines", updatePermission,
		middleware.SetAuditOperation("创建流水线"),
		cicdAPI.SavePipeline)
	g.PUT("/pipelines/:id", updatePermission, cicdAPI.SavePipeline)
	g.DELETE("/pipelines/:id", updatePermission,
		middleware.SetAuditOperation("删除流水线"),
		cicdAPI.DeletePipeline)
}

File: frontend/src/api/cicd.js
import request from './request'

// Jenkins Config
export const listJenkinsConfigs = (params) => request.get('/cicd/jenkins', { params })
export const saveJenkinsConfig = (data) => request.post('/cicd/jenkins', data)
export const updateJenkinsConfig = (id, data) => request.put(`/cicd/jenkins/${id}`, data)
export const deleteJenkinsConfig = (id) => request.delete(`/cicd/jenkins/${id}`)
export const testJenkinsConnection = (data) => request.post('/cicd/jenkins/test', data)

// Jobs
export const listJobs = (configId, params) => request.get(`/cicd/jenkins/${configId}/jobs`, { params })
export const triggerBuild = (configId, data) => request.post(`/cicd/jenkins/${configId}/build`, data)
export const listBuilds = (configId, params) => request.get(`/cicd/jenkins/${configId}/builds`, { params })
export const getBuildLog = (configId, params) => request.get(`/cicd/jenkins/${configId}/build-log`, { params })

// Pipelines
export const listPipelines = (params) => request.get('/cicd/pipelines', { params })
export const savePipeline = (data) => request.post('/cicd/pipelines', data)
export const updatePipeline = (id, data) => request.put(`/cicd/pipelines/${id}`, data)
export const deletePipeline = (id) => request.delete(`/cicd/pipelines/${id}`)
